addons/crm/static/src/js/crm_kanban.js

```diff
@@ -0,0 +1,44 @@
+odoo.define('crm.crm_kanban', function (require) {
+    "use strict";
+
+    var KanbanModel = require('web.KanbanModel');
+    var KanbanView = require('web.KanbanView');
+    var viewRegistry = require('web.view_registry');
+
+    var CrmKanbanModel = KanbanModel.extend({
+        moveRecord: async function (recordID, groupID, parentID) {
+            var result = await this._super(...arguments);
+            if (this.localData[parentID].groupedBy[0] === this.defaultGroupedBy[0]) {
```

Comment:
Maybe can you add a comment to explain that we are checking if the kanban view is grouped by "stage_id" ?